src/main/scala/loamstream/util/Processes.scala

```diff
@@ -0,0 +1,50 @@
+package loamstream.util
+
+import scala.collection.mutable.Buffer
+import scala.collection.mutable.ArrayBuffer
+import scala.util.Try
+import scala.util.Success
+
+/**
+ * @author clint
+ * May 17, 2018
+ */
+object Processes extends Loggable {
+  import scala.sys.process._
+  
+  def runSync(executable: String, commandLineTokens: Seq[String]): Try[RunResults] = {
+    import scala.sys.process._
+    
+    //NB: Implicit conversion to ProcessBuilder :\ 
+    val processBuilder: ProcessBuilder = commandLineTokens
+    
+    runSync(executable, processBuilder)
+  }
+  
+  def runSync(
+      executable: String, 
+      processBuilder: ProcessBuilder)(implicit logCtx: LogContext): Try[RunResults] = {
+    
+    val stdOutBuffer: Buffer[String] = new ArrayBuffer
+    val stdErrBuffer: Buffer[String] = new ArrayBuffer
+    
+    val processLogger = ProcessLogger(stdOutBuffer += _, stdErrBuffer += _)
+    
+    val attempt = Try {
+      val exitCode = processBuilder.!(processLogger)
+    
+      RunResults(executable, exitCode, stdOutBuffer.toList, stdErrBuffer.toList)
```

Comment:
Coming back to an earlier comment about `Try` wrapping a failed `RunResult`. It looks like getting the `exitCode` is what can fail. So, I wonder if there's just a better way of handling that case? For example:

```scala
val results = {
  val exitCode = Try(processBuilder.!(processLogger))

  // note: RunResults constructor take a Try[Int] for the exitCode parameter now
  RunResults(executable, exitCode, ...)
}

results match {
  case _ if runResults.isUnknown => ... // exitCode = Failure(..)
  case _ if runResults.isFailure => ...
  case _ => ...
}
```

That might make things easier?

Reply:
The line `processBuilder.!(processLogger)` runs the process - synchronously - collects its exit code, and invokes the appropriate method on `processLogger` once for each stdout and stderr line produced.  So if the call to `!` fails, there's basically no `RunResults` to return.  

I'll see what I can do to make things better here.  Maybe a new ADT that captures the states that need capturing here, instead of `Try`.